chapters/inheritance.tex

```diff
@@ -452,6 +461,43 @@ \subsection{Single Modification}\label{single-modification}
 m1(r = 1.6, r "x")
 m1(r = R(), r(y(min = 2)))
 \end{lstlisting}
+
+Modifiers can be merged for non-replaceable redeclarations, or replaceable redeclarations with a constraining type.
+\begin{lstlisting}[language=modelica]
+model Test
+  model A
+    replaceable Real x = 1;
+  end A;
+
+  A a(redeclare Real x, x.start = 2);
+  // Identical to A a(redeclare Real x(start=2));
+  A a(redeclare replaceable Real x constrainedby Real, x.start = 2);
+  // Identical to A a(redeclare Real x constrainedby Real(start=2));
+end Test;
+\end{lstlisting}
+
+For replaceable redeclarations without a constraining type the merging is only conceptual.
```

Comment:
This is where I typed a comment before but it disappeared. Trying again. I don't find that "only conceptual" explains enough what is going on. Can we formulate it more concretely? From the example it looks like there is still some merging going on. Is it correct then to say that if a redeclaration is also replaceable but does not include its own constranedby-clause, a constrainedby-clause is created using the class specified in the previous constrainedby-clause and the modification added to it? This topic is one of those where I forget the all the minute details if I haven't looked at it in more than a week.

Reply:
I have now added the forward reference and rewritten that section,  @eshmoylova